c/public/lib/source/ttread/ttread.c

```diff
@@ -820,21 +993,551 @@ static void hmtxRead(ttrCtx h) {
+    dnaDCL(boolean, bHasDelta);
+    dnaDCL(uint16_t, sharedPoints);
+    dnaDCL(uint16_t, privatePoints);
+    long pointIndicesCount;
+    uint16_t *pointIndices;
+    int nComponents = nPoints;
+    int i, j;
+    long k, l;
+    uint16_t hmtxPhantomCnt = 0;
+    Fixed scalar;
+    uint16_t variationDataSize;
+
+    if (gid >= h->gvar.dataOffsets.cnt)
+        fatal(h, ttrErrBadGlyphData, "no gvar data for gid [%d]", gid);
+
+    if (h->gvar.dataOffsets.array[gid] >= h->gvar.dataOffsets.array[gid + 1])
+        return; /* ignore if glyph variation data for this glyph is empty */
+
+    if (h->vf.flags & VF_FLAG_HMETRICS)
+        hmtxPhantomCnt = PHANTOM_HMTX_COUNT;
+
+    nPoints += PHANTOM_COUNT; /* add phantom points */
+    nTotalPoints += PHANTOM_COUNT; /* add phantom points */
+    dnaINIT(h->ctx.dna, xIntDeltas, nPoints, 0);
+    dnaINIT(h->ctx.dna, yIntDeltas, nPoints, 0);
+    dnaINIT(h->ctx.dna, sharedPoints, nPoints, 0);
+    dnaINIT(h->ctx.dna, privatePoints, nPoints, 0);
+    dnaINIT(h->ctx.dna, xFixedDeltas, nTotalPoints, 0);
+    dnaINIT(h->ctx.dna, yFixedDeltas, nTotalPoints, 0);
+    dnaINIT(h->ctx.dna, xDeltaSums, nTotalPoints, 0);
+    dnaINIT(h->ctx.dna, yDeltaSums, nTotalPoints, 0);
+    dnaINIT(h->ctx.dna, xOrig, nTotalPoints, 0);
+    dnaINIT(h->ctx.dna, yOrig, nTotalPoints, 0);
+    dnaINIT(h->ctx.dna, bHasDelta, nTotalPoints, 0);
+    dnaSET_CNT(xIntDeltas, nPoints);
+    dnaSET_CNT(yIntDeltas, nPoints);
+    dnaSET_CNT(sharedPoints, nPoints);
+    dnaSET_CNT(privatePoints, nPoints);
+    dnaSET_CNT(xFixedDeltas, nTotalPoints);
+    dnaSET_CNT(yFixedDeltas, nTotalPoints);
+    dnaSET_CNT(xDeltaSums, nTotalPoints);
+    dnaSET_CNT(yDeltaSums, nTotalPoints);
+    dnaSET_CNT(xOrig, nTotalPoints);
+    dnaSET_CNT(yOrig, nTotalPoints);
+    dnaSET_CNT(bHasDelta, nTotalPoints);
+    memset(xDeltaSums.array, 0, sizeof(Fixed) * nTotalPoints);
+    memset(yDeltaSums.array, 0, sizeof(Fixed) * nTotalPoints);
+    tupleHeaderOffset = h->gvar.tableOffset + h->gvar.dataArrayOffset + h->gvar.dataOffsets.array[gid];
+    srcSeek(h, tupleHeaderOffset);
+
+    tupleCount = read2(h);
+    dataOffset = (unsigned long)read2(h);
+    if (h->gvar.dataArrayOffset + dataOffset >= h->gvar.tableLength)
+        fatal(h, ttrErrBadGlyphData, "variation data offset for gid [%d] beyond gvar end", gid);
+
+    serializedDataOffset = tupleHeaderOffset + dataOffset;
+    tupleHeaderOffset = srcTell(h);
+
+    /* get the shared point numbers if present. */
+    if (tupleCount & gvar_FLAG_SHARED_POINT_NUMBERS) {
+        srcSeek(h, serializedDataOffset);
+        /* deltas are applied to ALL the points if points count is zero. */
+        pntCount = (unsigned long)read1(h);
+
+        if (pntCount == 0) {
+            bAllSharedPoints = 1;
+            pointIndicesCount = 0;
+        } else {
+            dnaSET_CNT(sharedPoints, gvarReadPackedPointNumbers(h, pntCount, &sharedPoints.array[0], nPoints));
+            bAllPoints = 0;
```

Comment:
The value stored in `bAllPoints` here is never read.

Reply:
correct. removed line